internal/notifier/alertmanager.go

```diff
@@ -137,15 +137,19 @@ func (s *Alertmanager) Post(ctx context.Context, event eventv1.Event) error {
 		},
 	}
 
-	var opts []requestOptFunc
+	opts := []postOption{
+		withProxy(s.ProxyURL),
```

Comment:
There's no point in making options optional if we always pass them. We need to check if the option is set and only pass it if it is.

Reply:
Looks like you understood it, basically I was talking about the `if` statements for adding `withSomething(...)` :+1: